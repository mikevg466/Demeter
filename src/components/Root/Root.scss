@import '../../stylesheets/modules/classNames';
@import '../../stylesheets/modules/colors';

.#{$project-name} {
  .root {
    height: calc(100vh - 80px);
    width: 100vw;
    padding-top: 80px;
    background-color: $dark-background;
    text-align: center;

    .header {
      padding-bottom: 100px;
      height: 40%;

      .logo {
        margin: 0 10%;
        height: 80%;
        @keyframes root-logo-spin {
          from {
            transform: rotate(0deg);
          }
          to {
            transform: rotate(-90deg);
          }
        }

        animation: root-logo-spin 3s ease-in-out 1s infinite alternate;

        .cheeks {
          fill: $girl-dark;
        }
        .swaddle-dark {
          fill: $boy-dark;
        }
        .swaddle-light {
          fill: $boy-light;
        }
      }
    }
  }
}

.#{$project-name} {
  .root.girl {
    .header {
      .logo {
        .swaddle-dark {
          fill: $girl-dark;
        }
        .swaddle-light {
          fill: $girl-light;
        }
      }
    }
  }
}
