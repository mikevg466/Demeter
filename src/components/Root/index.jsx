import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import GenderSelectorContainer from '../../containers/GenderSelector';
import DualSelectorContainer from '../../containers/DualSelector';
import FinalViewContainer from '../../containers/FinalView';
import ListPreviewContainer from '../../containers/ListPreview';
import Loader from '../../components/Loader';
import Logo from '../Icons/Baby';
import Constants from '../../utils/constants';
import './Root.scss';

/**
 * Root React Component containing the presentational Root
 * @extends PureComponent
 */
export default class RootComponent extends PureComponent {
  /**
   * React Render
   * @return {JSX}
   */
  render() {
    const {
      displayListPreview,
      genderType,
      isLoading,
      isSortingFinished,
      setGenderType,
    } = this.props;
    const isGenderSelected = !!genderType;

    const {
      GENDER_TYPES: { MALE, FEMALE },
    } = Constants;
    const genderClassName =
      genderType === MALE ? 'boy' : genderType === FEMALE ? 'girl' : '';

    return (
      <div className={`root ${genderClassName}`}>
        <header className='header'>
          <div
            className={`back-arrow ${isGenderSelected ? '' : 'hidden'}`}
            onClick={() => setGenderType('')}
          >
            {'<'}
          </div>
          <div className='logo-container'>
            <div className='logo'>
              <Logo />
            </div>
          </div>
        </header>
        {!isGenderSelected ? (
          <GenderSelectorContainer />
        ) : isLoading ? (
          <Loader />
        ) : isSortingFinished ? (
          <FinalViewContainer />
        ) : displayListPreview ? (
          <ListPreviewContainer />
        ) : (
          <DualSelectorContainer />
        )}
      </div>
    );
  }
}

RootComponent.propTypes = {
  displayListPreview: PropTypes.bool.isRequired,
  genderType: PropTypes.string.isRequired,
  isLoading: PropTypes.bool.isRequired,
  isSortingFinished: PropTypes.bool.isRequired,
  setGenderType: PropTypes.func.isRequired,
};
