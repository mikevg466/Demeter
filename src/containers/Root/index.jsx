import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { loadNames } from '../../actions/names';
import Root from '../../components/Root';

/**
 * Root React Container to be initialized in the main index.js file
 * @extends Component
 */
export class RootContainer extends Component {
  /**
   * React componentDidMount
   * @return {undefined}
   */
  componentDidMount() {
    const { loadNames } = this.props;
    loadNames();
  }

  /**
   * React Render
   * @return {JSX}
   */
  render() {
    const { isLoading } = this.props;
    console.log(isLoading);
    return <Root isLoading={isLoading} />;
  }
}

RootContainer.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  loadNames: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  isLoading: !state.names.list.length,
});

export const mapDispatchToProps = {
  loadNames,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps //eslint-disable-line
)(RootContainer);
