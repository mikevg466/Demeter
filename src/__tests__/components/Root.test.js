import React from 'react';
import { shallow } from 'enzyme';
import Root from '../../components/Root';

const mockProps = {
  displayListPreview: true,
  genderType: 'F',
  isLoading: false,
  isSortingFinished: true,
  setGenderType: jest.fn(),
};

describe('Root Component', () => {
  it('renders for girls', () => {
    const wrapper = shallow(<Root {...mockProps} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('renders for boys', () => {
    const wrapper = shallow(<Root {...mockProps} genderType='M' />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('renders with displayListPreview set to false', () => {
    const wrapper = shallow(<Root {...mockProps} displayListPreview={false} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('renders with isLoading set to true', () => {
    const wrapper = shallow(<Root {...mockProps} isLoading />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('renders with isSortingFinished set to false', () => {
    const wrapper = shallow(<Root {...mockProps} isSortingFinished={false} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
