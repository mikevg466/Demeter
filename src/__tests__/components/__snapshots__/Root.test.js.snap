// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Component renders for boys 1`] = `
"<div className=\\"root boy\\">
  <header className=\\"header\\">
    <div className=\\"back-arrow \\" onClick={[Function: onClick]}>
      &lt;
    </div>
    <div className=\\"logo-container\\">
      <div className=\\"logo\\">
        <BabyIcon />
      </div>
    </div>
  </header>
  <Connect(FinalViewContainer) />
</div>"
`;

exports[`Root Component renders for girls 1`] = `
"<div className=\\"root girl\\">
  <header className=\\"header\\">
    <div className=\\"back-arrow \\" onClick={[Function: onClick]}>
      &lt;
    </div>
    <div className=\\"logo-container\\">
      <div className=\\"logo\\">
        <BabyIcon />
      </div>
    </div>
  </header>
  <Connect(FinalViewContainer) />
</div>"
`;

exports[`Root Component renders with displayListPreview set to false 1`] = `
"<div className=\\"root girl\\">
  <header className=\\"header\\">
    <div className=\\"back-arrow \\" onClick={[Function: onClick]}>
      &lt;
    </div>
    <div className=\\"logo-container\\">
      <div className=\\"logo\\">
        <BabyIcon />
      </div>
    </div>
  </header>
  <Connect(FinalViewContainer) />
</div>"
`;

exports[`Root Component renders with isLoading set to true 1`] = `
"<div className=\\"root girl\\">
  <header className=\\"header\\">
    <div className=\\"back-arrow \\" onClick={[Function: onClick]}>
      &lt;
    </div>
    <div className=\\"logo-container\\">
      <div className=\\"logo\\">
        <BabyIcon />
      </div>
    </div>
  </header>
  <Loader />
</div>"
`;

exports[`Root Component renders with isSortingFinished set to false 1`] = `
"<div className=\\"root girl\\">
  <header className=\\"header\\">
    <div className=\\"back-arrow \\" onClick={[Function: onClick]}>
      &lt;
    </div>
    <div className=\\"logo-container\\">
      <div className=\\"logo\\">
        <BabyIcon />
      </div>
    </div>
  </header>
  <Connect(ListPreviewContainer) />
</div>"
`;
